### ISSUE TITLE
[java] WhileLoopWithLiteralBoolean - false negative with complex expressions

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3455]
**Affects PMD Version:6.35.0** 

**Rule:WhileLoopWithLiteralBoolean**

**Link:**
https://pmd.github.io/pmd-6.36.0/pmd_rules_java_bestpractices.html#whileloopwithliteralboolean

**Description:**
Hi, when using PMD, I found some false negatives caused by compound but constant expression. I have provided them in the following minimal code sample. Actually, expressions in conditional statements are constant and they should have been detected. Thanks.

**Code Sample demonstrating the issue:**

```java
public void func() {
    do {
      // Loop Body
    } while(false || false); // This is a false negative
//        } while(false | false); // This is a similar false negative.
}
```

**Expected outcome: Two warnings**

PMD should report two violations at line 4 and 5, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by VoidxHoshi** [comment]
I will have a try.

**Comment by Han-Cui** [comment]
When I use pmd 6.48.0 and 6.52.0, I still met this False Negative.

**Comment by adangel** [comment]
> When I use pmd 6.48.0 and 6.52.0, I still met this False Negative.

Thanks for letting us know. Can you please provide a complete test code sample that you think is not detected and create a new issue for that? Thanks!
